pkg/k8shandler/status.go

```diff
@@ -89,12 +93,35 @@ func getNodeStatus(name string, status *api.ElasticsearchStatus) (int, *api.Elas
 	return NOT_FOUND_INDEX, &api.ElasticsearchNodeStatus{}
 }
 
-func rolePodStateMap(namespace string, clusterName string) map[api.ElasticsearchNodeRole]api.PodStateMap {
+func rolePodStateMap(namespace, clusterName string, client client.Client) map[api.ElasticsearchNodeRole]api.PodStateMap {
 
-	baseSelector := fmt.Sprintf("component=elasticsearch,cluster-name=%v", clusterName)
-	clientList, _ := GetPodList(namespace, fmt.Sprintf("%s,%s", baseSelector, "es-node-client=true"))
-	dataList, _ := GetPodList(namespace, fmt.Sprintf("%s,%s", baseSelector, "es-node-data=true"))
-	masterList, _ := GetPodList(namespace, fmt.Sprintf("%s,%s", baseSelector, "es-node-master=true"))
+	clientList, _ := GetPodList(
+		namespace,
+		map[string]string{
+			"component":      "elasticsearch",
+			"cluster-name":   clusterName,
+			"es-node-client": "true",
+		},
+		client,
+	)
+	dataList, _ := GetPodList(
+		namespace,
+		map[string]string{
+			"component":    "elasticsearch",
```

Comment:
does it work? do we still use this lable?

Reply:
yes, we use `component=elasticsearch` label
from `common.go:L314 func newLabels(...)`:
```
	return map[string]string{
		"es-node-client":                   strconv.FormatBool(roleMap[api.ElasticsearchRoleClient]),
		"es-node-data":                     strconv.FormatBool(roleMap[api.ElasticsearchRoleData]),
		"es-node-master":                   strconv.FormatBool(roleMap[api.ElasticsearchRoleMaster]),
		"cluster-name":                     clusterName,
		"component":                        "elasticsearch",
		"tuned.openshift.io/elasticsearch": "true",
		"node-name":                        nodeName,
	}
```